packages/python-packages/api-stub-generator/apistub/nodes/_function_node.py

```diff
@@ -259,6 +279,21 @@ def add_error(self, error_msg):
             self.errors.append(error_msg)
 
 
+    def _validate_pageable_api(self):
+        # If api name starts with "list" and if annotated with "@distributed_trace"
+        # then this method should return ItemPaged or AsyncItemPaged
```

Comment:
We should also confirm that this is the return type in both the docstring and typehint - or will that already be validated by the check that compares the two?
Lastly - we should add this that this also confirms the presence of a subtype, e.g.
Not valid: `:rtype: ItemPaged`
Valid: `:rtype: ItemPaged[~azure.foo.Properties]`

Reply:
return type in docstring and typehint is already validated when parsing typehint itself. I will add a check to ensure docstring is wrapping actual return type with ItemPaged.

Reply:
@annatisch I have made change in REGEX to ensure return type is not just ItemPaged and actual type within wrapped ItemPaged is also part of it.